tests/components/husqvarna_automower_ble/test_config_flow.py

```diff
@@ -0,0 +1,192 @@
+"""Test the Husqvarna Bluetooth config flow."""
+
+from unittest.mock import Mock, patch
+
+from bleak import BleakError
+import pytest
+
+from homeassistant.components.husqvarna_automower_ble.const import DOMAIN
+from homeassistant.config_entries import SOURCE_BLUETOOTH, SOURCE_USER
+from homeassistant.const import CONF_ADDRESS, CONF_CLIENT_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import (
+    AUTOMOWER_MISSING_SERVICE_SERVICE_INFO,
+    AUTOMOWER_SERVICE_INFO,
+    AUTOMOWER_UNNAMED_SERVICE_INFO,
+    AUTOMOWER_UNSUPPORTED_GROUP_SERVICE_INFO,
+)
+
+from tests.components.bluetooth import inject_bluetooth_service_info
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+@pytest.fixture(autouse=True)
+def mock_random() -> Mock:
+    """Mock random to generate predictable client id."""
+    with patch(
+        "homeassistant.components.husqvarna_automower_ble.config_flow.random"
+    ) as mock_random:
+        mock_random.randint.return_value = 1197489078
+        yield mock_random
+
+
+async def test_user_selection(hass: HomeAssistant) -> None:
+    """Test we can select a device."""
+
+    inject_bluetooth_service_info(hass, AUTOMOWER_SERVICE_INFO)
+    inject_bluetooth_service_info(hass, AUTOMOWER_UNNAMED_SERVICE_INFO)
+    await hass.async_block_till_done(wait_background_tasks=True)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_ADDRESS: "00000000-0000-0000-0000-000000000001"},
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "confirm"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={},
+    )
+    assert result["title"] == "Husqvarna Automower"
+    assert result["result"].unique_id == "00000000-0000-0000-0000-000000000001"
+
+    assert result["data"] == {
+        CONF_ADDRESS: "00000000-0000-0000-0000-000000000001",
+        CONF_CLIENT_ID: 1197489078,
+    }
+
+
+async def test_no_devices(hass: HomeAssistant) -> None:
+    """Test missing device."""
+
+    inject_bluetooth_service_info(hass, AUTOMOWER_MISSING_SERVICE_SERVICE_INFO)
+    inject_bluetooth_service_info(hass, AUTOMOWER_UNSUPPORTED_GROUP_SERVICE_INFO)
+    await hass.async_block_till_done(wait_background_tasks=True)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
```

Comment:
What should this test test?

Reply:
In theory it's testing for a device with a missing `service_uuids`

I think I just copied this from the Gardena tests, and modified it for the AutoMower, so it might not be very relevant. But I thought it seemed useful to keep

Reply:
Yea but it's essentially testing nothing, it injects 2 services and then starts a user flow but it doesn't really do anything